pkg/driver/trace_driver.go

```diff
@@ -375,11 +603,11 @@ func (d *Driver) functionsDriver(list *list.List, announceFunctionDone *sync.Wai
 	waitForInvocations.Wait()
 
 	log.Debugf("All the invocations for function %s have been completed.\n", function.Name)
-	announceFunctionDone.Done()
-
 	atomic.AddInt64(totalSuccessful, successfulInvocations)
 	atomic.AddInt64(totalFailed, failedInvocations)
 	atomic.AddInt64(totalIssued, numberOfIssuedInvocations)
+	atomic.AddInt64(entriesWritten, functionsInvoked)
```

Comment:
What is the purpose of entriesWritten? Is this variable necessary?

Reply:
This variable counts the total number of functions that have been successfully invoked. Which will then be sent to the csv writer in line 887: ```totalIssuedChannel <- atomic.LoadInt64(&entriesWritten)```. 

In the previous single function implementation, 1 invocation of a function= 1 csv entry. This is counted using the variable ```totalIssued``` in the functionsDriver. 
However, in the DAG implementation: 1 invocation of the DAG = multiple function invocation, and multiple csv entries, depending on the size of the DAG. Thus we require an additional variable to count the total number of functions invoked.